### ISSUE TITLE
Enable `enable_data_node_lookup` by default

### ISSUE THREAD

**Issue by kaikash** [ytsaurus/ytsaurus#208]
`enable_data_node_lookup` helps to dramatically reduce network consumption, why is this parameter not enabled by default? it might be worth turning it on by default or at least adding it to the documentation.

**Comment by savrus** [comment]
By switching on enable_data_node_lookup you're disabling default block cache and start reading every key from disk. Not everybody is happy with such a behaviour by default. At last it is not a good idea to switch default value

As for the documentation, we'll add the description.

**Comment by kaikash** [comment]
What about network consumption? In my case setting `enable_data_node_lookup ` reduced network consumption by 20 times: lookup is ~3GiB/100k rps and `chunk_reader_statistics_data_bytes_transmitted` is about 60GiB. What are the recommendations reducing network usage?

**Comment by kaikash** [comment]
And about block cache, is it enabled by default?
`yt get //sys/cluster_nodes/@config/%true/data_node` returns no `block_cache` section, does it mean block cache is disabled?

**Comment by Quid37** [comment]
Hi!

> What about network consumption? 

Indeed data node lookup is one of (though quite peculiar) ways to tackle problems with high network usage.

> returns no block_cache section

This yson represents dynamic configuration of cluster nodes, could you please check your node static configuration files as well?
You can also observe block cache utilization with chunk_reader_statistics/data_bytes_read_from_cache sensors.

**Comment by kaikash** [comment]
What are the other options that could help reducing network consumption?
In my case, the table has 1 light column, which is looked up by the user, and a heavy column (megabytes). As far as I understand, such lines lie in one block and that is why there is such a large network consumption.

**Comment by gritukan** [comment]
`What are the other options that could help reducing network consumption?
In my case, the table has 1 light column, which is looked up by the user, and a heavy column (megabytes). As far as I understand, such lines lie in one block and that is why there is such a large network consumption.`

Do you need a value of the heavy column in the lookup? If not, you can just use `optimize_for=scan` to turn on columnar format that puts values of different columns into different blocks.

**Comment by kaikash** [comment]
No, don't need heavy column in these lookups and it is not looked up neither. Will try to set `optimize_for=scan`. The main load falls on the lookups and changing `optimize_for` from `lookup` to `scan` seems to be a little counterintuitive.  What are the nuances, except that there will more disk seeks when more columns are needed?

**Comment by Quid37** [comment]
You will also need to specify `group` attribute for desired columns of your schema because each block stores columns of a single group only. Unless specified each column in stored in a common group.

It may get a little counterintuitive with `optimize_for` options, however this is more about the way of storing data (row-wise vs column-wise). On practice scan format can get noticeably faster due to better compression and block cache utilization.

**Comment by Quid37** [comment]
As for block cache. It shall be set to 1Gb by default.
You can increase its size in dynamic config by setting the field `data_node/block_cache/uncompressed_data/capacity` to new value in bytes.

We will soon write an optimization guide where these options and features will be covered.
